[Prachi Goenka] ip

prachi2023 opened the pull request.

prachi2023 committed:
Level 0. Greet

Add a hello greeting
Add exit greeting

prachi2023 committed:
add user input, add echoing of commands

prachi2023 committed:
Add list function. Add display of list of text entered

prachi2023 committed:
Add tracking of completion of tasks Add function to mark tasks as done

prachi2023 committed:
Add method for execute and determining commands sent

prachi2023 committed:
Add subclass for deadline, event and todo

Edit the main to take to be able to add these tasks to the list of tasks

prachi2023 committed:
Edit the test case to test the level - 4 version of duke

prachi2023 committed:
Edit the executeCommand method and addTask to follow SLAP. Refactored some methods like addTask

prachi2023 committed:
Update the runtest file to be able to show the ticks and crosses. Changed the expected text to contain the ticks and crosses

wangqinNick commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,12 @@
+public class Deadline extends Task {
```
Good job for avoiding the code violation

wangqinNick commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+    public static Task[] tasks = new Task[10];
+
+    public static void exitDuke (){
+        shouldExit = true;
```
avoid using "should" "must" in the code.

wangqinNick reviewed (commented):
Generally good job on avoiding the code violation!

yujinyang1998 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,12 @@
+        this.by = by;
+    }
+
+    public String toString (){
```
Could add a comment on the rough output of the string here to increase readability.

yujinyang1998 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+    }
+
+    // Mark a specific task in the list as done
+    public static void markTaskInListAsDone(String command){
```
Naming for function is abit too long. Perhaps shorten it a little

yujinyang1998 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,89 @@
+            command = in.nextLine();
+            executeCommand(command);
+        }
     }
```
Good job in extracting out the classes to make the code more readable. Keep it up!

prachi2023 committed:
Handle exception for getting the int index of the task to be marked as done

prachi2023 committed:
Handle the exception for an invalid add task command

prachi2023 committed:
Handle exceptions while adding tasks
exceptions handled include:
command entered is not understood
no description given after the command
for deadlines and events no /at or /by was entered

prachi2023 committed:
Add DukeExceptions class

prachi2023 committed:
Organize the classes into packages. duke.task

prachi2023 committed:
Merge branch 'branch-A-Packages'

prachi2023 committed:
Edit the exception handling to use DukeException

prachi2023 committed:
Merge branch 'branch-Level-5'

# Conflicts: Resolve conflict due to package naming
#	src/main/java/duke/Duke.java

prachi2023 committed:
Add public getIsDone method for Task

prachi2023 committed:
Add DukeException class into a package
import the package into the main duke file

prachi2023 committed:
Change to storage of the tasks from array to ArrayList

prachi2023 committed:
Add delete task function

prachi2023 committed:
Edit delete function to check and show if the tasks list is empty

prachi2023 committed:
Add function to determine if file exists otherwise create a txt file called tasklist

prachi2023 committed:
Add function to add todo functions to the file

prachi2023 committed:
Add a method to convert the task to string for the format to be placed on the file

prachi2023 committed:
Add method to read and load data already there on the file into duke

prachi2023 committed:
Add method to be able the file when a task gets updated
When a task changes from being not done to done --> the file updates

prachi2023 committed:
Add delete task method

prachi2023 committed:
Merge branch 'branch-Level-6'

# Conflicts:
#	src/main/java/duke/Duke.java

prachi2023 committed:
Solve merge conflict with branch 7 and master

prachi2023 committed:
Edit to be able to delete task from file
rearrange the methods to make it more readable

prachi2023 committed:
change file path

leowweixiang commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,19 @@
+package duke.exception;
+
+public class DukeException extends Exception {
+    public DukeException (String error){
```
I'll recommend keeping this code clean by simply creating the Exception with the error message, instead of some pseudo-enum that has to be processed further.

If you're concerned about certain exceptions being repeated often and so you don't want to have magic messages everywhere, then my recommendation is to at least not pass in a String parameter, but create an Enum that maps to specific messages, e.g. EXCEPTION_NODESCRIPT etc. This allows you to know for sure what error you're talking about, and avoid things like typo mistakes, e.g. DukeException("no description ")

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+        String command = userInput[0];
+        // Determine the command
+        switch (command) {
+        case "bye":
```
Recommend creating these as constants on top. That way you are very clear what kind of commands you expect instead of having to navigate this section of code

leowweixiang commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,19 @@
+    public DukeException (String error){
+        switch (error) {
+        case "no description":
+            System.out.println("Oops it seems you have left the description empty");
```
Also don't recommend that the exception prints the error message, but let the catcher deal with it. For example, you print a message here, but in the above code you catch and print a completely independent message again, which defeats the point of throwing

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+        if (taskDetails[0].isEmpty()){
+            throw new DukeException("no description");
+        }
+        if (!task.contains("/at")){
```
Recommend checking this before splitting on "/at"

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+
+    /* Saving file and Editing file */
+    //Check and create file
+    public static void setUpFile() throws IOException{
```
These file operations, do you want them in the Duke class?

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+
+    // Mark a specific task in the list as done
+    public static void markTaskInListAsDone(int taskIndexCompleted){
+            if (taskIndexCompleted > tasks.size()) {
```
Should be 4-space indentation

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+        if (tasks.isEmpty()){
+            System.out.println("There is no task to delete");
+        }
+        else if (taskToDelete > tasks.size()) {
```
Consider dealing with this with a common Java exception

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+    // Print the list of tasks entered
+    public static void printList (){
```
Same for all these. Your Duke is pretty bloated right now, so it is good to move things to different classes

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+
+public class Duke {
+    public static boolean shouldExit = false;
+    public static ArrayList<Task> tasks = new ArrayList<>();
```
Is there a need for these to be static?

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,259 @@
+public class Duke {
+    public static boolean shouldExit = false;
+    public static ArrayList<Task> tasks = new ArrayList<>();
+    public static String path = "tasklist.txt";
```
Since it is a constant, should be final as well

prachi2023 committed:
Extract ui into its own package and class

prachi2023 committed:
Extract Command, Parser and TaskList class

prachi2023 committed:
Edit Duke Exception to initialise with an error message

prachi2023 committed:
Extract storage into a new class

prachi2023 committed:
Add a find Command class

prachi2023 committed:
Edit the existing classes to, TaskList, Parser and Ui to find keyword

prachi2023 committed:
Modify the deadline class to take the date as a LocalDate instead of a string. Updated parser to add a function that parses the string into the date. Uodate relevant methos to take in LocalDate instead of string when printing and adding a new deadline

prachi2023 committed:
Add datetime features to events

prachi2023 committed:
Add the ability to list tasks based on the month, year and date

prachi2023 committed:
Merge pull request #2 from prachi2023/branch-Level-9

Branch level 9

prachi2023 committed:
Merge branch 'master' of https://github.com/prachi2023/ip

* 'master' of https://github.com/prachi2023/ip:
  Edit the existing classes to, TaskList, Parser and Ui to find keyword
  Add a find Command class

prachi2023 committed:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/task/Task.java
#	tasklist.txt

prachi2023 committed:
Merge pull request #1 from prachi2023/branch-Level-8

Branch level 8

prachi2023 committed:
Add a help command to show all available commands

prachi2023 committed:
Fix data of tasks when printing lists

prachi2023 committed:
Add JavaDocs to half of the public methods and classess

prachi2023 committed:
Merge branch 'branch-A-Javadoc'

prachi2023 committed:
Add more JavaDocs

prachi2023 committed:
Edit file creatio error

prachi2023 committed:
Edit ReaMe to include feature list

prachi2023 committed:
Edit markdown to include feature descriptions and command summary

prachi2023 committed:
Set theme jekyll-theme-midnight

prachi2023 committed:
Set theme jekyll-theme-hacker

prachi2023 committed:
Set theme jekyll-theme-cayman

prachi2023 committed:
Update README.md

prachi2023 committed:
Update README.md

prachi2023 committed:
Update ReadMe
Fix a bug in todo that would not show when 'today' command is entered

prachi2023 committed:
Remove issue where if no keyword or index is entered there is an unhandled exception

prachi2023 committed:
Exxtract method from deadline and event case in parser
